### DISCUSSION TITLE
Which inverse transform to apply while using Numpyro models with Blackjax?

### DISCUSSION THREAD

**Discussion by Madhav-Kanda** [blackjax-devs/blackjax#556] [Q&A]
The Numpyro [example](https://blackjax-devs.github.io/blackjax/examples/howto_use_numpyro.html) provided mentions that:

`The model applies a transformation to the theta variable. As a result, the samples generated by Blackjax will be samples in the transformed space and you will have to transform them back to the original space with Numpyro.`

How can we know which inverse transformation we must apply to get the parameters back into the constrained space?

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

When using a NumPyro model the sampler runs in the unconstrained space, so apply the model's own inverse transform to get back to the constrained space. Use the `postprocess_fn` from `numpyro.infer.util.initialize_model` (or `biject_to(constraint)`), `vmap`-ed over your samples. This is the same mechanism as logging deterministics — see #792.
